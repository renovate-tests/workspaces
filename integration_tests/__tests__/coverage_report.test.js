/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @flow
 */
'use strict';

const fs = require('fs');
const path = require('path');
const skipOnWindows = require('../../scripts/skip_on_windows');
const {extractSummary} = require('../utils');
const runJest = require('../runJest');

const DIR = path.resolve(__dirname, '../coverage_report');

skipOnWindows.suite();

test('outputs coverage report', () => {
  const {stdout, status} = runJest(DIR, ['--no-cache', '--coverage']);
  const coverageDir = path.resolve(__dirname, '../coverage_report/coverage');

  // - the `setup.js` file is ignored and should not be in the coverage report.
  // - `sum_dependency.js` is mocked and the real module is never required but
  //  is listed with 0 % coverage.
  // - `not-required-in-test-suite.js` is not required but it is listed
  //  with 0 % coverage.
  expect(stdout).toMatchSnapshot();

  expect(() => fs.accessSync(coverageDir, fs.F_OK)).not.toThrow();
  expect(status).toBe(0);
});

test('collects coverage only from specified files', () => {
  const {stdout} = runJest(DIR, [
    '--no-cache',
    '--coverage',
    '--collectCoverageFrom', // overwrites the one in package.json
    'setup.js',
  ]);

  // Coverage report should only have `setup.js` coverage info
  expect(stdout).toMatchSnapshot();
});

test('collects coverage only from specified files avoiding dependencies', () => {
  const {stdout} = runJest(DIR, [
    '--no-cache',
    '--coverage',
    '--collectCoverageOnlyFrom',
    'sum.js',
    '--',
    'sum.test.js',
  ]);

  // Coverage report should only have `sum.js` coverage info
  expect(stdout).toMatchSnapshot();
});

test('json reporter printing with --coverage', () => {
  const {stderr, status} = runJest('json_reporter', ['--coverage']);
  const {summary} = extractSummary(stderr);
  expect(status).toBe(1);
  expect(summary).toMatchSnapshot();
});

test('outputs coverage report as json', () => {
  const {stdout, status} = runJest(DIR, ['--no-cache', '--coverage', '--json']);
  expect(status).toBe(0);

  try {
    JSON.parse(stdout);
  } catch (err) {
    throw new Error(
      "Can't parse the JSON result from stdout. " + err.toString(),
    );
  }
});
