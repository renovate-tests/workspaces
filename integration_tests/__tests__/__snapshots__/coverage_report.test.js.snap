// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collects coverage only from specified files 1`] = `
"----------|----------|----------|----------|----------|----------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------|----------|----------|----------|----------|----------------|
All files |      100 |      100 |      100 |      100 |                |
 setup.js |      100 |      100 |      100 |      100 |                |
----------|----------|----------|----------|----------|----------------|
"
`;

exports[`collects coverage only from specified files avoiding dependencies 1`] = `
"----------|----------|----------|----------|----------|----------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------|----------|----------|----------|----------|----------------|
All files |    85.71 |      100 |       50 |    85.71 |                |
 sum.js   |    85.71 |      100 |       50 |    85.71 |             13 |
----------|----------|----------|----------|----------|----------------|
"
`;

exports[`json reporter printing with --coverage 1`] = `
"Test Suites: 1 failed, 1 total
Tests:       1 failed, 2 passed, 3 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`outputs coverage report 1`] = `
"-------------------------------|----------|----------|----------|----------|----------------|
File                           |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-------------------------------|----------|----------|----------|----------|----------------|
All files                      |    41.18 |        0 |       25 |    41.18 |                |
 not-required-in-test-suite.js |        0 |        0 |        0 |        0 |  9,16,17,18,20 |
 other-file.js                 |      100 |      100 |      100 |      100 |                |
 sum.js                        |    85.71 |      100 |       50 |    85.71 |             13 |
 sum_dependency.js             |        0 |        0 |        0 |        0 |     9,11,12,15 |
-------------------------------|----------|----------|----------|----------|----------------|
"
`;
